### DISCUSSION TITLE
Concatenate widget doesn't name Source Identification attributes, or how to name a data table?

### DISCUSSION THREAD

**Discussion by Bigfoot-solutions** [biolab/orange3#6326] [Q&A] [answered]
I am trying to concatenate a set of 20 datasets and use the "Append data source ID" option to retain visibility into which data element came from which input source. 

The issue I am running into is that even though the input data tables have unique names in the UI (Unit A logs, Unit X logs, etc) the data info widget still displays the name of every data table as "untitled", and the Concatenate widget appends source ID's of "untitled (0)" and "untitled (1)" apparently based on the order in which the tables were connected to the Concatenate widget. I have a work-around using the Create Class widget to rename the Source ID, but this is very brittle. 

Am I doing something wrong or should this be a bug report? Is there a mechanism for naming data tables that I can't find in the documentation? 

Thanks for any help.

**Comment by janezd** [comment]
Can you perhaps share the workflow? You can omit the data if you can't share it.

**Reply by Bigfoot-solutions** [reply]
I can't share too much of the workflow but I pared it down to a simple example that shows the issue and the workaround I am using. 

[Data Analysis workflow snippet.zip](https://github.com/biolab/orange3/files/10583329/Data.Analysis.workflow.snippet.zip)

**Comment by janezd** [comment]
Where do you assign names to tables?

Note that the name of the File widget is not the name of the table. The name is, initially the file name, but may change later. Manually, you can do it in the Edit Domain widget.

**Reply by Bigfoot-solutions** [reply]
Great to know!

**Comment by markotoplak** [comment, marked as answer]
Thanks. Your problem was likely fixed by @PrimozGodec in #6331
